source/agora/consensus/data/Transaction.d

```diff
@@ -97,6 +110,10 @@ public struct Transaction
 
     public void deserialize (scope DeserializeDg dg) nothrow @safe
     {
+        TxType type_val;
+        deserializePart(type_val, dg);
```

Comment:
You don't need to declare a temporary anymore. You can write `deserializePart(this.type, dg);`

Reply:
Suggested change